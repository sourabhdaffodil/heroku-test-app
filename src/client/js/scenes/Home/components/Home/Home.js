import React, { Component } from 'react';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { getAbout, getOrganizations, getHistory } from '../../../../actions';
import { media } from '../../../../theme/style-utils';
import Header from '../Header/Header';
import Subhead from '../Subhead/Subhead';
import Navigation from '../Navigation/Navigation';
import CommitHistory from '../CommitHistory/CommitHistory';
import Organizations from '../Organizations/Organizations';
import Footer from '../Footer/Footer';
import { fetchAbout } from '../../services/about';
import { fetchOrganizations } from '../../services/organizations';
import { fetchHistory } from '../../services/history';

const HomeContainer = styled.div`
  font-family: ${props => props.theme.fontFamily('primary')};
  display: flex;
  flex-direction: column;
  height: 100%;
`;

const MainContent = styled.div`
  max-width: 1012px;
  width: 100%;
  box-sizing: border-box;
  padding: 0 12px;
  margin: 0 auto;
  flex-grow: 1;

  ${media.phone`
    padding: 0;
  `}
`;

const Desc = styled.p`
  ${props => props.theme.textSetting('nr')};

  ${media.phone`
    display: none;
  `}
`;

class Home extends Component {
  componentDidMount() {
    fetchAbout().then((response) => {
      this.props.getAbout(response);
    });
    fetchOrganizations().then((response) => {
      this.props.getOrganizations(response);
    });
    fetchHistory().then((response) => {
      this.props.getHistory(response);
    });
  }

  render() {
    return (
      <HomeContainer>
        <Header />
        <Subhead />
        <Router>
          <div>
            <MainContent>
              <Desc>{this.props.description}</Desc>
              <Navigation />
              <Route exact path="/" component={CommitHistory} />
              <Route exact path="/commits" component={CommitHistory} />
              <Route exact path="/orgs" component={Organizations} />
              <Route exact path="/repos" component={CommitHistory} />
            </MainContent>
          </div>
        </Router>
        <Footer />
      </HomeContainer>
    );
  }
}

Home.propTypes = {
  getAbout: PropTypes.func.isRequired,
  getOrganizations: PropTypes.func.isRequired,
  getHistory: PropTypes.func.isRequired,
  description: PropTypes.string.isRequired,
};

function matchDispatchToProps(dispatch) {
  return bindActionCreators({
    getAbout,
    getOrganizations,
    getHistory,
  }, dispatch);
}

function mapStateToProps(state) {
  return {
    description: state.about.description,
  };
}

export default connect(mapStateToProps, matchDispatchToProps)(Home);

export {
  Home,
};
