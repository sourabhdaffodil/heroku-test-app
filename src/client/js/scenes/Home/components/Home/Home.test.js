import React from 'react';
import { shallow } from 'enzyme';
import { Home } from './Home';

describe('Home component', () => {
  it('Must render properly', () => {
    const container = shallow(
      <Home
        getAbout={() => null}
        getOrganizations={() => null}
        getHistory={() => null}
        description=""
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
