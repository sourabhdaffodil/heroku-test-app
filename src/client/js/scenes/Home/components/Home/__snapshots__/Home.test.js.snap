// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    description=""
    getAbout={[Function]}
    getHistory={[Function]}
    getOrganizations={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header />,
        <Connect(Subhead) />,
        <BrowserRouter>
          <div>
            <styled.div>
              <styled.p>
                
              </styled.p>
              <Connect(Navigation) />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/commits"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/orgs"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/repos"
              />
            </styled.div>
          </div>
        </BrowserRouter>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <styled.div>
              <styled.p>
                
              </styled.p>
              <Connect(Navigation) />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/commits"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/orgs"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/repos"
              />
            </styled.div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <styled.div>
              <styled.p>
                
              </styled.p>
              <Connect(Navigation) />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/commits"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/orgs"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/repos"
              />
            </styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.p>
                  
                </styled.p>,
                <Connect(Navigation) />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/commits"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/orgs"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/repos"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/commits",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/orgs",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/repos",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header />,
          <Connect(Subhead) />,
          <BrowserRouter>
            <div>
              <styled.div>
                <styled.p>
                  
                </styled.p>
                <Connect(Navigation) />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/commits"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/orgs"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/repos"
                />
              </styled.div>
            </div>
          </BrowserRouter>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <styled.div>
                <styled.p>
                  
                </styled.p>
                <Connect(Navigation) />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/commits"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/orgs"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/repos"
                />
              </styled.div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <styled.div>
                <styled.p>
                  
                </styled.p>
                <Connect(Navigation) />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/commits"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/orgs"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/repos"
                />
              </styled.div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.p>
                    
                  </styled.p>,
                  <Connect(Navigation) />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/commits"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/orgs"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/repos"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/commits",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/orgs",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/repos",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
